pyposeidon/mjigsaw.py

```diff
@@ -506,37 +506,38 @@ def get(contours, **kwargs):
         # ---------------------------------
 
         # execute jigsaw
-        with subprocess.Popen(
-            args=["jigsaw {}".format(tag + ".jig")],
+        ex = subprocess.run(
+            ["jigsaw {}".format(tag + ".jig")],
+            check=False,
+            capture_output=True,
+            text=True,
             cwd=calc_dir,
             shell=True,
-            stderr=subprocess.PIPE,
-            stdout=subprocess.PIPE,
-            #            bufsize=1,
-        ) as ex:
-            output = ex.stdout.read()
-            error = ex.stderr.read()
-
-        if error:
-            logger.error("Jigsaw FAILED\n")
-            with open(calc_dir + "err.log", "w") as f:
-                for line in error.splitlines():
-                    f.write(line.decode(sys.stdout.encoding))
-                    logger.debug(line.decode(sys.stdout.encoding))
+            universal_newlines=True,
+            # bufsize=1,
+        )
```

Comment:
Similar to before, I would rename `ex` to `proc` or `jigsaw_proc`, use `shlex.split()` and remove the `shell=True`. 

For the record, `universal_newlines` is not needed when you have `text=True`